db/view/view.cc

```diff
@@ -1391,9 +1391,9 @@ static std::optional<gms::inet_address>
 get_view_natural_endpoint(const sstring& keyspace_name,
         const dht::token& base_token, const dht::token& view_token) {
     auto &db = service::get_local_storage_proxy().local_db();
-    auto& topology = service::get_local_storage_proxy().get_token_metadata_ptr()->get_topology();
     auto& ks = db.find_keyspace(keyspace_name);
     auto erm = ks.get_effective_replication_map();
+    auto& topology = erm->get_token_metadata_ptr()->get_topology();
```

Comment:
The patch is okay, but since we drop erm after returning the result, it could be out of date when used. Probably it will be fixed later on in the series.

Reply:
not in this series, that's focused on repair.
this is a localized cleanup to just get the topology from the erm that's is already retrieved here.

Reply:
I think we need to decouple mv updates from storage_proxy, or we'll get into a difficult circular dependency problem. That is, the mv update is a new sub-transaction with its own erm.